puppetboard/templates/nodes.html

```diff
@@ -1,34 +1,64 @@
 {% extends 'layout.html' %}
 {% import '_macros.html' as macros %}
 {% block content %}
-<div class="ui fluid icon input hide" style="margin-bottom:20px">
-  <input autofocus="autofocus" class="filter-table" placeholder="Type here to filter...">
+<div class="ui grid">
+  <div class="fourteen wide column">
+    <div class="ui fluid icon input">
+      <input id="main-table-search" autofocus="autofocus" type="text" placeholder="Type here to filter...">
+      <i class="search icon"></i>
+    </div>
+  </div>
+  <div class="two wide column">
+    <div class="ui fluid icon dropdown large basic button">
+      <input name="status" type="hidden" value="Failed">
+      <i class="filter icon"></i>
+      <span class="text">Filter By Status</span>
+      <i class="dropdown icon"></i>
+      <div class="menu">
+        <a class="item" href="{{url_for_field('status', 'failed')}}">Failed</a>
+        <a class="item" href="{{url_for_field('status', 'changed')}}">Changed</a>
+        <a class="item" href="{{url_for_field('status', 'unchanged')}}">Unchanged</a>
+        <a class="item" href="{{url_for_field('status', 'noop')}}">Noop</a>
+        <a class="item" href="{{url_for_field('status', 'unreported')}}">Unreported</a>
+      </div>
+    </div>
+  </div>
 </div>
-<table class='ui very compact very basic sortable table'>
+<table
+  id="main-table"
+  class='ui very basic very compact table'
+  data-auto-width="false"
+  data-paging="false"
+  data-info="false"
+  data-dom="t"
+  data-order='[[ 2, "asc" ]]'
```

Comment:
The current default sort order is descending - from nodes with the newest reports to the oldest - and I think that we should keep it that way.

Can you change it, please?


Also, even if you change the order to descending, you still get the "unreported" nodes above the reported ones.

Example:

<img width="1326" alt="Screenshot 2022-08-29 at 17 18 21" src="https://user-images.githubusercontent.com/566632/187248227-aafc366d-10e5-47b4-907b-4ca287b52792.png">

This is also a different behavior than what we had and I don't think it's beneficial. (Firstly, it's not logical as the unreported nodes are not reported because they have been seen by the PuppetDB too long time ago. Secondly, it's just not practical as if you want to look at the nodes with the latest catalogs then these are not among them. Also their list may be a long one so it will take a lot of time to scroll through these.)

Can we change that too, please?

Reply:
@gdubicki I made the change that you ask. I made a default to empty string for column ordering when no timestamp and i have done the same render behavior as Report column.